phenotrex/ml/prediction.py

```diff
@@ -1,22 +1,34 @@
+import pandas as pd
+
 from phenotrex.io.flat import load_genotype_file, DEFAULT_TRAIT_SIGN_MAPPING
 from phenotrex.io.serialization import load_classifier
+from phenotrex.ml.shap_handler import ShapHandler
+
 try:
     from phenotrex.transforms import fastas_to_grs
 except ModuleNotFoundError:
     from phenotrex.util.helpers import fail_missing_dependency as fastas_to_grs
 
 
-def predict(fasta_files=tuple(), genotype=None, classifier=None, verb=False):
+def predict(fasta_files=tuple(), genotype=None, classifier=None,
+            out_explain_per_sample=None, out_explain_summary=None,
+            n_max_explained_features=None, verb=False):
     """
     Predict phenotype from a set of (possibly gzipped) DNA or protein FASTA files
-    or a single genotype file.
+    or a single genotype file. Optionally, compute SHAP explanations individually and/or summarily
+    for the predicted samples.
+
     NB: Genotype computation is highly expensive and performed on the fly on FASTA files.
     For increased speed when predicting multiple phenotypes, create a .genotype file to reuse
     with the command `compute-genotype`.
 
+    NB: Computing SHAP explanations on SVM models is highly costly.
```

Comment:
If possible, could you give some rough estimate, like "10x slower than plain SVM" or similar?

Reply:
It depends on how deeply one chooses to sample with the KernelExplainer, but I would assume for reasonable parameters, I would estimate a range from several seconds per sample, to several hundreds of seconds per sample.